src/runtime/analytics-service.js

```diff
@@ -62,18 +65,30 @@ export class AnalyticsService {
      */
     this.context_ = new AnalyticsContext();
 
-    /**
-     * @private @const {!TransactionId}
-     */
-    this.xid_ = new TransactionId(deps);
-
     /** @private {?Promise<!web-activities/activity-ports.ActivityIframePort>} */
     this.serviceReady_ = null;
 
     /** @private {?Promise} */
     this.lastAction_ = null;
   }
 
+  /**
+   * @param {string} transactionId
+   * @param {boolean=} override
+   */
+  setTransactionId(transactionId, override = false) {
+    if (override) {
+      this.transactionId_ = transactionId;
```

Comment:
It also needs to set `this.context_.setTransactionId(...)` to the new value. This would need a test.

Reply:
Done.